yarn-project/noir-contracts/scripts/types.sh

```diff
@@ -22,14 +22,27 @@ ROOT=$(pwd)
 
 write_import() {
     CONTRACT_NAME=$1
-    NAME=`echo $CONTRACT_NAME | sed -r 's/(^|_)(.)/\U\2/g'`
+
+    if [ "$(uname)" = "Darwin" ]; then
+        # sed \U doesn't work on mac
+        NAME=$(echo $CONTRACT_NAME | perl -pe 's/(^|_)(\w)/\U$2/g')
+    else
+        NAME=`echo $CONTRACT_NAME | sed -r 's/(^|_)(.)/\U\2/g'`
+    fi
+
     echo "import ${NAME}Json from './${CONTRACT_NAME}_contract.json' assert { type: 'json' };"  >> "$artifacts_dir/index.ts";
 }
 
 write_export() {
     CONTRACT_NAME=$1
-    NAME=`echo $CONTRACT_NAME | sed -r 's/(^|_)(.)/\U\2/g'`
-    
+    if [ "$(uname)" = "Darwin" ]; then
+        # sed \U doesn't work on mac
```

Comment:
Can't we just always use perl for simplicity?

Reply:
yep don't know why I didn't think of that 😬

Reply:
not supported in CI :(